README.md

```diff
@@ -104,12 +102,29 @@ tree -L 1 ~/code/exercism
 
 #### Regenerating a Test Suite
 
-From within the ruby directory, run the following command:
+From time to time, the [canonical data](https://github.com/exercism/problem-specifications/tree/master/exercises) for an exercise's tests changes, and we need to keep the Ruby version's tests synced up.  Regenerating these tests is a quick and easy way to help maintain the track and get involved!
 
-    bin/generate --update <slug> 
+If it's your first time cloning/contributing to the repository, you'll need to install any dependencies via `bundle`:
 
-Leaving out the --update option will cause the BookKeeping version number to remain the same.
-This can be useful when testing generators.
+```bash
+bundle install
+```
+
+The generator also depends on the presence of Exercism's `problem-specifications` repository (see the file tree in the section above).  Make sure you've got an *up-to-date* version of the specifications in a `problem-specifications` folder that's in a parallel directory to your local copy of the `ruby` repository.
+
+To check which problems have possibly been updated, run:
+
+```bash
+bin/generate -a
+```
+
+This will autogenerate all of the exercises that can be.  Use your editor, diff tool, or `git status` to find out which test files have changed.  Some exercises will update because someone updated the description or other exercise metadata.  Others will change because the actual test suite has changed.  If you find that an exercise's test suite (i.e. the actual tests, not just the line at the test data version number at the top of the tests) has changed, be sure to use `generate` to auto-increment the [BookKeeping::VERSION](#bookkeeping-version) number by running:
```

Comment:
> all of the exercises that can be.

all of the test cases for which generators exist.

Reply:
> Use your editor, diff tool, or `git status` to find out which test files have changed.

Too many options here. 
`git diff` is a useful command to use to see what has changed.

Reply:
> to auto-increment

to update